release/ray_release/util.py

```diff
@@ -34,6 +39,35 @@ def __str__(self):
 ]
 
 
+def get_read_state_machine_aws_bucket(allow_pr_bucket: bool = False) -> str:
+    # We support by default reading from the branch bucket only, since most of the use
+    # cases are on branch pipelines. Changing the default flag to read from the bucket
+    # according to the current pipeline
+    if allow_pr_bucket:
+        return get_write_state_machine_aws_bucket()
+    return (
+        get_global_config()["state_machine_aws_bucket"]
+        or get_global_config()["state_machine_branch_aws_bucket"]
+    )
+
+
+def get_write_state_machine_aws_bucket() -> str:
+    # We support different buckets for writing test result data; one for pr and one for
+    # branch. This is because pr and branch pipeline have different permissions, and we
+    # want data on branch pipeline being protected.
+    bucket_v1 = get_global_config()["state_machine_aws_bucket"]
+    if bucket_v1:
+        return bucket_v1
+
+    pipeline_id = os.environ.get("BUILDKITE_PIPELINE_ID")
+    if pipeline_id in BRANCH_PIPELINES:
+        return get_global_config()["state_machine_branch_aws_bucket"]
+    if pipeline_id in PR_PIPELINES:
+        return get_global_config()["state_machine_pr_aws_bucket"]
+
+    assert False, f"Test state machine is not supported for pipeline {pipeline_id}"
```

Comment:
just assert `pipeline_id in PR_PIPELINES` on line 65?

`assert False` feels like a misuse on assertion.